cilium/CFP-26135-default-egress-allow.md

```diff
@@ -0,0 +1,198 @@
+
+
+ * Option 1 is preferred as it aligns with existing code flow and does not affect existing network policies schema or behavior in any way.
+
+#### Adding new field in ruleMetadata
+Based on config received via command line, Cilium Agent would convert it to EgressDeny rule and does policyAdd. PolicyAdd will add the rules to repository and triggers policyreaction event to trigger identity allocation for labels specified in the policy. Cilium by default goes to deny mode if any policy added. To prevent this, this rule can be created with this flag `enableDefaultAllow` enabled. If a rule has this flag enabled, then cilium will never enforce default deny for the rest of traffic unless if there any egress k8s network policy or cilium network policy rule is enforced. By default, this flag is disabled and it would not regress existing network policy scenarios.
+
+`api.Rule` can extend `ruleMetadata` in `api:rule` structure to add a new field named `enableDefaultAllow`. When evaluating matching policy rules for an endpoint, can differentiate this deny rule from other network policy rules based on this flag. If all matching rules `enableDefaultAllow` to true then default allow will be enforced. If any one of matching rule contains `enableDefaultAllow` to false, then default deny will be enforced.
+
+`Resource` field in `policy.AddOptions` used to indicate the source of policy rule. A new entry of `ResourceKind` named `ResourceKindDaemonDenyTuples` can be added to indicate the source of this rule is via commandLine.
+
+#### Examples
+##### Example 1
+
+* Cloud Provider deny tuples 
+-–deny-tuples=[“169.254.128.5;80;;]
+
+* Network Policy/Cilium Network Policy 
+None
+
+* Action
+Egress - Default Allow for all. Block access to 169.254.128.5:80 for all
+Ingress - Default Allow for all
+
+##### Example 2
+
+* Cloud Provider deny tuples 
+```-–d-eny-tuples=[“169.254.128.5;80;;]```
+
+* Network Policy/Cilium Network Policy 
+```
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  name: "l3-egress"
+spec:
+  endpointSelector:
+    matchLabels:
+      role: backend
+  egress:
+  - toEndpoints:
+    - matchLabels:
+        role: frontend
+```
+* Action
+Egress - Allow for all except for pods with label “role: backend”. For those pods, egress allowed only to pods with “role: frontend”. Block access to 169.254.128.5:80 for all
+Ingress - Default Allow for all
+
+
+#### Alternate Options
+#### Option 2 (CCNP CRD with new field)
```

Comment:
nit: If the subsequent options are nested, they should be one step deeper in the document hierarchy. This helps readers to sort through and understand the relationship of this section in comparison to the others. 
```suggestion
### Alternate Options
#### Option 2 (CCNP CRD with new field)
```